src/types/any.rs

```diff
@@ -87,6 +87,14 @@ impl PyAny {
         <T as PyTryFrom>::try_from(self)
     }
 
+    /// Returns whether the object is the same as the other.
+    ///
+    /// This is equivalent to the Python expression `self is other`.
```

Comment:
I think this is worth explaining in a bit more detail, because this can be pretty surprising to people unfamiliar with Python. And even people that should know better get `==` and `is` wrong sometimes.

```suggestion
    /// Returns whether `self` and `other` point to the same object. To compare the equality of two objects
    /// (the `==` operator), use [`eq`](PyAny::eq).
    ///
    /// This is equivalent to the Python expression `self is other`.
```
(the link needs the other PR to work)